src/dialogs/more-info/more-info-related-info.ts

```diff
@@ -0,0 +1,133 @@
+import { HassEntity } from "home-assistant-js-websocket";
+import { css, html, LitElement, TemplateResult } from "lit";
+import { customElement, property } from "lit/decorators";
+import { styleMap } from "lit/directives/style-map";
+import memoizeOne from "memoize-one";
+import { fireEvent } from "../../common/dom/fire_event";
+import { computeStateDisplay } from "../../common/entity/compute_state_display";
+import { stateColor } from "../../common/entity/state_color";
+import { stateIconPath } from "../../common/entity/state_icon_path";
+import "../../components/ha-chip";
+import "../../components/ha-chip-set";
+import "../../components/ha-icon";
+import { HomeAssistant } from "../../types";
+
+declare global {
+  interface HASSDomEvents {
+    "related-entity": { entityId: string };
+  }
+}
+
+@customElement("more-info-related-info")
+class MoreInfoContent extends LitElement {
+  @property({ attribute: false }) public hass?: HomeAssistant;
+
+  @property({ attribute: false }) public stateObj?: HassEntity;
+
+  private _relatedEntities = memoizeOne((entityId: string) => {
+    const deviceId = this.hass!.entities[entityId].device_id;
+
+    if (!deviceId) return [];
+
+    return Object.values(this.hass!.entities).filter(
+      (entity) =>
+        entity.device_id === deviceId &&
+        entity.entity_id !== entityId &&
+        !entity.hidden_by &&
+        !entity.disabled_by
+    );
+  });
+
+  private _handleChipClick(ev: Event) {
+    const entityId = (ev.target as any).entityId as string;
+    fireEvent(this, "related-entity", { entityId });
+  }
+
+  protected render(): TemplateResult | null {
+    if (!this.hass || !this.stateObj) {
+      return null;
+    }
+
+    const relatedEntities = this._relatedEntities(this.stateObj.entity_id);
+
+    return html`
+      <div class="container">
+        ${relatedEntities.map((entry) => {
+          const entity = this.hass!.states[entry.entity_id];
+
+          const icon = entity.attributes.icon;
+          const iconPath = stateIconPath(entity);
+          const state = computeStateDisplay(
+            this.hass!.localize,
+            entity,
+            this.hass!.locale
+          );
+          const color = stateColor(entity);
+
+          return html`
+            <button
```

Comment:
We should use `ha-chip` here

Reply:
It's not the same design between the figma file and `ha-chip` so I created a custom class as a first version. May be we have to add additional style to `ha-chip`.

Reply:
New chips are not available in mdc lib : https://m3.material.io/components/chips/overview#5294ef21-85ac-404a-ae55-d3e7a065a390. It's just "planned".

Reply:
We should not have different  version floating around, we either change all the styling of chips, or add it with the old design until we update the styling of chips

Reply:
> New chips are not available in mdc lib : https://m3.material.io/components/chips/overview#5294ef21-85ac-404a-ae55-d3e7a065a390. It's just "planned".

Hmm, can we add our own style to it like we did with our buttons? Might be a different PR.